File: deepgoal/core/agents/impls.py
from __future__ import annotations

from ..node import NodeResult
from ..types import EngineOptions
from .base import BaseAgent


class AnalysisAgent(BaseAgent):
    async def execute(self, input_file_path: str, options: EngineOptions) -> NodeResult:
        print(f"[AnalysisAgent] {input_file_path} -> {self.output_file_path}")
        return NodeResult(success=True, output_file_path=self.output_file_path)


class DecompositionAgent(BaseAgent):
    async def execute(self, input_file_path: str, options: EngineOptions) -> NodeResult:
        print(f"[DecompositionAgent] {input_file_path} -> {self.output_file_path}")
        return NodeResult(success=True, output_file_path=self.output_file_path)


class PlanningAgent(BaseAgent):
    async def execute(self, input_file_path: str, options: EngineOptions) -> NodeResult:
        print(f"[PlanningAgent] {input_file_path} -> {self.output_file_path}")
        return NodeResult(success=True, output_file_path=self.output_file_path)


class TaskAgent(BaseAgent):
    async def execute(self, input_file_path: str, options: EngineOptions) -> NodeResult:
        print(f"[TaskAgent] {input_file_path} -> {self.output_file_path}")
        return NodeResult(success=True, output_file_path=self.output_file_path)


class ExecutorAgent(BaseAgent):
    async def execute(self, input_file_path: str, options: EngineOptions) -> NodeResult:
        print(f"[ExecutorAgent] {input_file_path} -> {self.output_file_path}")
        return NodeResult(success=True, output_file_path=self.output_file_path)


class AcceptanceAgent(BaseAgent):
    async def execute(self, input_file_path: str, options: EngineOptions) -> NodeResult:
        print(f"[AcceptanceAgent] {input_file_path} -> {self.output_file_path}")
        return NodeResult(success=True, output_file_path=self.output_file_path)

File: deepgoal-ts/src/core/agents/impls.ts
import { BaseAgent } from './base.js';
import { okResult, pipelineOutputFromPath, type PipelineInput, type PipelineOutput, type PipelineResult } from '../node.js';
import type { EngineOptions } from '../types/index.js';

abstract class NamedStubAgent extends BaseAgent {
  constructor(
    private readonly agentName: string,
    output?: PipelineOutput
  ) {
    super(output);
  }

  override async execute(input: PipelineInput, _options: EngineOptions): Promise<PipelineResult> {
    return okResult({
      ...this.output,
      metadata: {
        ...this.output.metadata,
        agent: this.agentName,
        input: input.primaryPath ?? '',
      },
    });
  }
}

export class AnalysisAgent extends NamedStubAgent {
  constructor(output?: PipelineOutput) {
    super('AnalysisAgent', output);
  }
}

export class DecompositionAgent extends NamedStubAgent {
  constructor(output?: PipelineOutput) {
    super('DecompositionAgent', output);
  }
}

export class PlanningAgent extends NamedStubAgent {
  constructor(output?: PipelineOutput) {
    super('PlanningAgent', output);
  }
}

export class TaskAgent extends NamedStubAgent {
  constructor(output?: PipelineOutput) {
    super('TaskAgent', output);
  }
}

export class ExecutorAgent extends NamedStubAgent {
  constructor(output?: PipelineOutput) {
    super('ExecutorAgent', output);
  }
}

export class AcceptanceAgent extends NamedStubAgent {
  constructor(output?: PipelineOutput) {
    super('AcceptanceAgent', output);
  }
}

export function outputPath(path: string): PipelineOutput {
  return pipelineOutputFromPath(path);
}
